### PR TITLE
mpich: fix macos problem with -flat-namespace

### PR DESCRIPTION
**Pull request by albestro**
See discussion in #27429, it is just applying what has been decided there.

I had to create a new one due to a mess I did with the branch on my fork.

### CODE REVIEW

**Comment by alalazo** [comment]
@eschnett fyi

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/mpich/package.py [hunk 1]
```diff
@@ -491,6 +482,10 @@ def configure_args(self):
             "--with-yaksa={0}".format(spec["yaksa"].prefix if "^yaksa" in spec else "embedded"),
         ]
 
+        # see https://github.com/pmodels/mpich/issues/5530
+        if sys.platform == "darwin":
```
Comment (haampie) [inline var/spack/repos/builtin/packages/mpich/package.py]:
```suggestion
        if spec.platform == "darwin":
```

**Review by haampie — DISMISSED** [review]

**Review by alalazo — APPROVED** [review]

**Review by raffenet — DISMISSED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by naromero77** [comment]
Can someone check why the CI pipeline failed?

**Comment by naromero77** [comment]
@spackbot commands

**Comment by naromero77** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Review by raffenet — COMMENTED** [review]
My only concern is that this may cause issues with older versions. We only switched the default in MPICH 4.1. Maybe add a condition on the version being built?

**Pull request merged**